libs/SalesforceSDK/src/com/salesforce/androidsdk/analytics/SalesforceAnalyticsManager.java

```diff
@@ -70,8 +70,10 @@ public class SalesforceAnalyticsManager {
     private static final String UNAUTH_INSTANCE_KEY = "_no_user";
 
     private static Map<String, SalesforceAnalyticsManager> INSTANCES;
-    private static boolean sPublishHandlerActive;
-    private static int sPublishFrequencyInHours = DEFAULT_PUBLISH_FREQUENCY_IN_HOURS;
+    private static boolean isPublishWorkRequestEnqueued;
+    private static boolean isPublishOneTimeOnAppBackgroundEnabled = true;
```

Comment:
Do we want to support isPublishOneTimeOnAppBackgroundEnabled and isPublishPeriodicallyOnFrequencyEnabled both flags being true (or false). Should we instead have an enum with three values no-publishing, publish-on-background, publish-periodically?

Reply:
I like this idea.

Reply:
I _also_ like that idea. I'll get a follow-up work item open to review feedback from @wmathurin also.